src/helper/QJsonObjectManipulation.h

```diff
@@ -25,7 +25,7 @@ inline std::string convertToString(QJsonObject jsonData) {
 
 //converts unix timestamp to human-readable in YYYY-MM-DD format (the rest of seconds is neglected)
 //needed to call the api about market news. Will be called in api.h
-inline std::string convertToReadable(long unixTimeStamp) {
+inline std::string convertToReadable(long long unixTimeStamp) {
```

Comment:
This change does not compile on my system.

@ngkan Can we remove this change from this PR and instead transition API and Stock classes to use QDateTime (https://doc.qt.io/qt-5/qdatetime.html#currentSecsSinceEpoch) instead of std::time ?

@makni-mehdi The problem stems from the fact that we use https://en.cppreference.com/w/cpp/chrono/c/time. std::time(0) can return either 32-bit (long int) or 64-bit (long long int) based on the system (not the compiler)

But I believe transitioning to QDateTime should be done to make this system independent and work everywhere. What do you think?

Reply:
I was not aware of QDateTime. I will have a  deeper look at documentation and change it today if possible. Indeed we want it to work properly everywhere.